aquasec/resource_application_scope.go

```diff
@@ -335,10 +338,10 @@ func resourceApplicationScopeCreate(d *schema.ResourceData, m interface{}) error
 
 	if err == nil {
 		d.Set("categories", flattenCategories(iap.Categories))
-
+		d.SetId(name)
 		err1 := resourceApplicationScopeRead(d, m)
 		if err1 == nil {
-			d.SetId(name)
+
```

Comment:
please change it to if err1 != true {return fmt.Errorf("application scope resource read is failed with error: %v", err1)}

Reply:
have pushed a change to improve this code